src/cursor/abstract_cursor.ts

```diff
@@ -836,7 +852,7 @@ function makeCursorStream<TSchema extends Document>(cursor: AbstractCursor<TSche
         // NOTE: This is also perhaps questionable. The rationale here is that these errors tend
         //       to be "operation interrupted", where a cursor has been closed but there is an
         //       active getMore in-flight.
-        if (cursor.killed) {
+        if (err.message.match(/interrupted/)) {
```

Comment:
Is this a safe check? Maybe we want to capture rational in a comment here

Reply:
The cursor was previously only killed on cursor.close() but it changed to be killed during cleanup. Which meant that legitimate errors were not causing the stream to destroy because at this point the cursor was _always_ killed.

Reply:
Also updated the comment in 363d41b22523c6324693f92ea6f6db88c5b85458